### PR TITLE
added test for BaseSerializerHandler

### PR DESCRIPTION
**Pull request by ahmetakbn**
### Subject

Improving tests for `SonataCoreBundle`

### CODE REVIEW

File: Tests/Serializer/BaseSerializerHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
```
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Doesn't this test feel weird? If yes, it is probably because the formats are hardcoded in the class. Improve that if you feel up to it (by introducing a `formats` property that you might change from your tests), and it the end, the tests should look like this : 
- check that there are twice as many methods as the number of formats
- check that half for them are about deserialization…

File: Tests/Serializer/BaseSerializerHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        $visitor = $this->getMock('JMS\Serializer\VisitorInterface');
+        $visitor->expects($this->once())
+            ->method('visitInteger')
+            ->with(1, array('foo'), $context)
+            ->willReturn(true);
+
+        $serializer = new FooSerializer($manager);
+
+        $this->assertTrue($serializer->serializeObjectToId($visitor, $data, array('foo'), $context));
```
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: How does this test check that we have an instance of manager ?

File: Tests/Serializer/BaseSerializerHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+        $context = $this->getMock('JMS\Serializer\Context');
+
+        $visitor = $this->getMock('JMS\Serializer\VisitorInterface');
+        $visitor->expects($this->never())
+            ->method('visitInteger');
+
+        $serializer = new FooSerializer($manager);
+
+        $serializer->serializeObjectToId($visitor, $data, array('foo'), $context);
```
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: There is a lot of repetition here. Could you please use a data provider or a private method ?

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 1 above)
Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: While writing the test for this class, I was aware that the formats were hardcoded. It is indeed weird. But since I was writing test, this was another concern. I don't want to change the structure of the class while writing test for it. This should be business of another PR IMO. After adding the property the tests also must be updated as the new structure requires.

Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: > if you feel up to it

Sure, I understand perfectly, no problem :)

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 2 above)
Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: I am manuplating the return for `getClass` method of `$manager` to an instance of my fake class `FooSerializer`. And I set the variable `$data` to instance of my fake class as well. 'Data is Instance of Manager' is the logic in the method but test does not work with exact instance of manager of course.

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 3 above)
Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: What is the repetition exactly I didn't get it. The mocks are a bit different then the mock in previous test method.

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 1 above)
Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: So there is one more thing to do :)

File: Tests/Serializer/BaseSerializerHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
+    }
+
+    public function testSerializeObjectToIdWithDataIsInstanceOfManager()
+    {
+        $data = $this->getMock(
```
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Please find a more descriptive name than `$data`

Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Sure, I can. But this is the name of the original method's parameter.

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 3 above)
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Indeed, I read a bit too fast :)

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Ok

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 4 above)
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: Oh god…

Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: I'd say rename it anyway. If someone wants to change that in the class someday, they probably won't think of looking at the tests too…

**Comment by Soullivaneuh** [comment]
@greg0ire I let you handle this PR.

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 4 above)
Comment (ahmetakbn) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: ok, I changed the name to `$modelInstance`

File: Tests/Serializer/BaseSerializerHandlerTest.php (hunk 1 above)
Comment (greg0ire) [inline Tests/Serializer/BaseSerializerHandlerTest.php]: See #306

**Comment by greg0ire** [comment]
Thanks a lot @ahmetakbn !
